presto-main/src/main/java/com/facebook/presto/resourcemanager/DistributedClusterStatsResource.java

```diff
@@ -45,18 +50,38 @@ public class DistributedClusterStatsResource
     private final boolean isIncludeCoordinator;
     private final ResourceManagerClusterStateProvider clusterStateProvider;
     private final InternalNodeManager internalNodeManager;
+    private final boolean isClusterStatsCacheEnabled;
+    private Supplier<ClusterStats> clusterStatsSupplier;
 
     @Inject
-    public DistributedClusterStatsResource(NodeSchedulerConfig nodeSchedulerConfig, ResourceManagerClusterStateProvider clusterStateProvider, InternalNodeManager internalNodeManager)
+    public DistributedClusterStatsResource(
+            NodeSchedulerConfig nodeSchedulerConfig,
+            ResourceManagerConfig resourceManagerConfig,
+            ServerConfig serverConfig,
+            ResourceManagerClusterStateProvider clusterStateProvider,
+            InternalNodeManager internalNodeManager)
     {
         this.isIncludeCoordinator = requireNonNull(nodeSchedulerConfig, "nodeSchedulerConfig is null").isIncludeCoordinator();
         this.clusterStateProvider = requireNonNull(clusterStateProvider, "nodeStateManager is null");
         this.internalNodeManager = requireNonNull(internalNodeManager, "internalNodeManager is null");
+        Duration expirationDuration = requireNonNull(serverConfig, "serverConfig is null").getClusterStatsExpirationDuration();
+        this.isClusterStatsCacheEnabled = expirationDuration.getValue() > 0;
+        if (isClusterStatsCacheEnabled) {
+            this.clusterStatsSupplier = memoizeWithExpiration(this::calculateClusterStats, expirationDuration.toMillis(), MILLISECONDS);
+        }
     }
 
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     public Response getClusterStats()
+    {
+        if (isClusterStatsCacheEnabled) {
+            return Response.ok(clusterStatsSupplier.get()).build();
+        }
+        return Response.ok(calculateClusterStats()).build();
+    }
```

Comment:
I think you can remove `isClusterStatsCacheEnabled` altogether by just checking if the constructor if the duration is 0--if it's 0, just set `clusterStatsSupper = this::caclulateClusterStats`, if non-zero then decorate it with the memoizer.